src/util/windows.jl

```diff
@@ -0,0 +1,66 @@
+"""
+An iterator which moves across a container, as a sliding window.
+"""
+immutable EachWindowIterator{T <: AbstractArray}
```

Comment:
Yeah, I think we can make the `Sequence` a subtype of `AbstractArray`. But I feel the `Sequence` is more like a subtype of `AbstractString` than `AbstractArray`, though I never see clear semantic distinction between them. At the moment, this type constraint would be too stringent unless the Julia community reaches a consensus about semantics of these abstract types.